core/src/main/java/org/elasticsearch/common/xcontent/support/filtering/FilterPathBasedFilter.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.xcontent.support.filtering;
+
+import com.fasterxml.jackson.core.filter.TokenFilter;
+import org.elasticsearch.common.util.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FilterPathBasedFilter extends TokenFilter {
+
+    /**
+     * Marker value that should be used to indicate that a property name
+     * or value matches one of the filter paths.
+     */
+    private final static TokenFilter MATCHING = new TokenFilter(){};
```

Comment:
That's just a minor thing but I think the recommended order in the Java styleguide is `private static final`.